New Entry: General Linux & Unix (#837)
Codecademy/docs: content/general/concepts/unix/unix.md

---
Title: 'Unix'
Description: 'Unix is a family of operating systems that are all derived from the original AT&T Unix developed at Bell Labs in 1969.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:

>   - 'Developer Tools'
>   - 'Operating Systems'

Reviewer: ```suggestion
  - 'Developer Tools'
  - 'Linux'
  - 'Operating Systems'
  - 'Unix'
```

CatalogContent:
  - 'paths/computer-science'
  - 'paths/code-foundations'
---

Unix is a family of operating systems that are all derived from the original AT&T Unix developed at Bell Labs in 1969. Unix is a multitasking, multiuser operating system that is characterized by the implementation of many small modular software tools that can be combined using a scripting language known as the [Unix shell](https://en.wikipedia.org/wiki/Unix_shell). Unix and Unix-like operating systems are currently used widely on servers, workstations and mobile devices.

Originally developed for internal use at Bell labs, in the late 1970s AT&T licensed Unix to a variety of outside parties, leading to a number of academic and commercial Unix variants. These include:

> - BSD: From the University of California, Berkley.
> - SunOS/Solaris: From Sun Microsystems.
> - AIX: From IBM

Reviewer: ```suggestion
- [Berkely Software Distribution](https://en.wikipedia.org/wiki/Berkeley_Software_Distribution#External_links) from the University of California, Berkley (now discontinued).
- [SunOS/Solaris](https://www.oracle.com/solaris/solaris11/) from Sun Microsystems (now Oracle).
- [Advanced Interactive eXecutive (AIX)](https://www.ibm.com/it-infrastructure/power/os/aix) from IBM.
```

> In 1991, [Linus Torvalds](https://www.codecademy.com/resources/docs/general/linus-torvalds) released the [open source](https://www.codecademy.com/resources/docs/general/open-source) [Linux](https://www.codecademy.com/resources/docs/general/linux) kernel as a Unix-like free alternative to proprietary Unix systems. It had since exploded into dozens, if not hundreds of different variant distributions.

Reviewer: ```suggestion
## Timeline

In 1991, [Linus Torvalds](https://www.codecademy.com/resources/docs/general/linus-torvalds) released the [open source](https://www.codecademy.com/resources/docs/general/open-source) [Linux](https://www.codecademy.com/resources/docs/general/linux) kernel as a Unix-like free alternative to proprietary Unix systems. It had since exploded into dozens, if not hundreds of different variant distributions.
```

In 1992, a free derivative of BSD Unix was released, leading to the [NetBSD](https://www.netbsd.org/) and [FreeBSD](https://www.freebsd.org/) projects. Since 1994 BSD Unix has developed several different branches, including [OpenBSD](https://www.openbsd.org/).

> In 2000 Apple released a BSD Unix System named Darwin that eventually became the core of the macOS operating system.

Reviewer: ```suggestion
In 2000 Apple released a BSD Unix System named [Darwin](https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#External_links) that eventually became the core of the macOS operating system.
```

---
Title: 'Unix'
Description: 'Unix is a family of operating systems that are all derived from the original AT&T Unix developed at Bell Labs in 1969.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Developer Tools'
  - 'Linux'
  - 'Operating Systems'
  - 'Unix'
CatalogContent:
  - 'paths/computer-science'
  - 'paths/code-foundations'
---

> **Unix** is a family of operating systems that are all derived from the original AT&T Unix developed at Bell Labs in 1969. Unix is a multitasking, multiuser operating system that is characterized by the implementation of many small modular software tools that can be combined using a scripting language known as the [Unix shell](https://en.wikipedia.org/wiki/Unix_shell). Unix and Unix-like operating systems are currently used widely on servers, workstations and mobile devices.

Reviewer 2: ```suggestion
**Unix** is a family of operating systems that are all derived from the original AT&T Unix developed at Bell Labs in 1969. Unix is a multitasking, multiuser operating system that is characterized by the implementation of many small modular software tools that can be combined using a scripting language known as the [Unix shell](https://en.wikipedia.org/wiki/Unix_shell). Unix and Unix-like operating systems are currently used widely on servers, workstations, and mobile devices.
```

> Originally developed for internal use at Bell labs in the late 1970s, AT&T licensed Unix to a variety of outside parties, leading to a number of academic and commercial Unix variants. These include:

Reviewer 2: ```suggestion
Originally developed for internal use at Bell Labs in the late 1970s, AT&T licensed Unix to a variety of outside parties, leading to a number of academic and commercial Unix variants. These include:
```

- [Berkely Software Distribution](https://en.wikipedia.org/wiki/Berkeley_Software_Distribution#External_links) from the University of California, Berkley (now discontinued).
- [SunOS/Solaris](https://www.oracle.com/solaris/solaris11/) from Sun Microsystems (now Oracle).
- [Advanced Interactive eXecutive (AIX)](https://www.ibm.com/it-infrastructure/power/os/aix) from IBM.

## Timeline

In 1991, [Linus Torvalds](https://www.codecademy.com/resources/docs/general/linus-torvalds) released the [open source](https://www.codecademy.com/resources/docs/general/open-source) [Linux](https://www.codecademy.com/resources/docs/general/linux) kernel as a Unix-like free alternative to proprietary Unix systems. It had since exploded into dozens, if not hundreds of different variant distributions.

In 1992, a free derivative of BSD Unix was released, leading to the [NetBSD](https://www.netbsd.org/) and [FreeBSD](https://www.freebsd.org/) projects. Since 1994 BSD Unix has developed several different branches, including [OpenBSD](https://www.openbsd.org/).

> In 2000 Apple released a BSD Unix System named [Darwin](https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#External_links) that eventually became the core of the macOS operating system.

Reviewer 2: ```suggestion
In 2000, Apple released a BSD Unix System named [Darwin](https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#External_links) that eventually became the core of the macOS operating system.
```
